test/server/connection_handler_test.cc

```diff
@@ -526,6 +528,27 @@ TEST_F(ConnectionHandlerTest, DisableListener) {
   handler_->disableListeners();
 }
 
+TEST_F(ConnectionHandlerTest, StopAndDisableStoppedListener) {
```

Comment:
What would cause Envoy to stop or disable an already stopped listener? Please add a comment for the test explaining.

Reply:
Actually, we don't have such case in the current code(as I known), I just make it works in case we have such in the future. 
In tcp listener, there is check to keep the thing safe 
https://github.com/envoyproxy/envoy/blob/35d18c932a325bf62a67b2d95021a18e0e32d167/source/server/active_tcp_listener.cc#L112-L116

But udp listener hasn't such check
https://github.com/envoyproxy/envoy/blob/35d18c932a325bf62a67b2d95021a18e0e32d167/source/server/active_udp_listener.h#L97

So I add the check at ConnectionhandleImpl.

Or I should using the assertion and cleanup the tcp and udp listener check?

Reply:
I understand now. I think this is fine as is.